app/models/match.rb

```diff
@@ -25,16 +25,59 @@
 #
 class Match < ApplicationRecord
   acts_as_tenant(:organization)
-  belongs_to :pet
+  belongs_to :pet, touch: true
   belongs_to :adopter_foster_account
 
+  has_one :user, through: :adopter_foster_account
+  has_one :adopter_foster_profile, through: :adopter_foster_account
+
+  validates :start_date,
+    presence: {if: -> { match_type == "foster" }}
+  validates :end_date,
+    presence: {if: -> { match_type == "foster" }},
+    comparison: {
+      if: -> { !!start_date },
+      greater_than: :start_date, message: "must be after %{value}"
+    }
+
   validate :belongs_to_same_organization_as_pet, if: -> { pet.present? }
-  validates :pet, uniqueness: true
 
   after_create_commit :send_reminder
 
   enum :match_type, [:adoption, :foster]
 
+  scope :fosters, -> { where(match_type: :foster) }
+
+  def self.foster_statuses
+    ["complete", "upcoming", "current"]
+  end
+
+  def self.ransackable_attributes(auth_object = nil)
+    %w[status]
+  end
+
+  def self.ransackable_associations(auth_object = nil)
+    %w[pet user]
+  end
+
+  def fosterer_name(format = :default)
+    user.full_name(format)
+  end
+
+  def status
+    return :not_applicable if start_date.nil? || end_date.nil?
```

Comment:
Anyone have thoughts on this method? Currently, it's only really applicable for fosters but could see a future where it could be useful for adoptions too.